spec/requests/api/docs/v3/declarations_spec.rb

```diff
@@ -0,0 +1,64 @@
+require "rails_helper"
+require "swagger_helper"
+
+RSpec.describe "Participant Declarations endpoint", type: :request, openapi_spec: "v3/swagger.yaml" do
+  include_context "with authorization for api doc request"
+
+  it_behaves_like "an API index endpoint documentation",
+                  "/api/v3/participant-declarations",
+                  "Participant declarations",
+                  "Participant declarations",
+                  "#/components/schemas/ListParticipantDeclarationsFilter",
+                  "#/components/schemas/ParticipantDeclarationsResponse"
+
+  describe "single declarations" do
+    let(:lead_provider) { create(:lead_provider) }
+    let(:type) { "participant-declaration" } # check
+    let(:application) { create(:application, :accepted, :with_declaration, lead_provider:) }
+    let(:resource) { application.declarations.first }
+
+    it_behaves_like "an API show endpoint documentation",
+                    "/api/v3/participant-declarations/{id}",
+                    "Participant declarations",
+                    "Participant declarations",
+                    "#/components/schemas/ParticipantDeclarationResponse"
+
+    it_behaves_like "an API update endpoint documentation",
+                    "/api/v3/participant-declarations/{id}/void",
+                    "Participant declarations",
+                    "Void a declaration",
+                    "The participant declaration being voided",
+                    "#/components/schemas/ParticipantDeclarationResponse"
+  end
+
+  describe "create declarations" do
+    let(:lead_provider) { create(:lead_provider) }
+    let(:type) { "participant-declaration" } # check
+    let(:cohort) { create(:cohort, :current) }
+    let(:course_group) { CourseGroup.find_by(name: "leadership") }
+    let(:course) { create(:course, :sl, course_group:) }
+    let!(:schedule) { create(:schedule, :npq_leadership_autumn, course_group:, cohort:) }
+    let(:application) { create(:application, :accepted, cohort:, course:, lead_provider:) }
+    let(:declaration_date) { schedule.applies_from + 1.day }
+    let(:response_example) do
+      extract_swagger_example(schema: "#/components/schemas/ParticipantDeclarationResponse", version: :v3)
```

Comment:
You can omit these if you aren't writing a custom response example, it will default to the `ParticipantDeclarationResponse` passed in on line 61

Reply:
Addressed in https://github.com/DFE-Digital/npq-registration/pull/1504/commits/4e0af42776d143d12a68a3f599bcb2a982136461